apps/src/templates/sectionProgress/SectionProgressToggle.jsx

```diff
@@ -27,62 +28,48 @@ class SectionProgressToggle extends React.Component {
   };
 
   state = {
+    initialView: ViewType.SUMMARY,
     selectedToggle: this.props.currentView
   };
 
   componentWillReceiveProps(nextProps) {
     // currentView can be set externally, and the component will
     // still need to update when that happens.
     if (this.state.selectedToggle !== nextProps.currentView) {
-      this.setState({selectedToggle: nextProps.currentView});
+      this.setState({
+        selectedToggle: nextProps.currentView,
+        initialView: nextProps.currentView
+      });
     }
   }
 
-  onChange = () => {
+  onChange = selectedToggle => {
+    firehoseClient.putRecord(
+      {
+        study: 'teacher_dashboard_actions',
+        study_group: 'progress',
+        event: 'view_change_toggle',
+        data_json: JSON.stringify({
+          section_id: this.props.sectionId,
+          old_view: this.state.initialView,
+          new_view: selectedToggle
+        })
+      },
+      {includeUserId: true}
+    );
     // Display the toggle based on the internal state so that it is
     // more immediately responsive. Once setting internal state is
     // complete, then update the redux currentView.
     // Timeouts forces a render of the local state before dispatching
     // the action.
-    if (this.state.selectedToggle === ViewType.SUMMARY) {
-      firehoseClient.putRecord(
-        {
-          study: 'teacher_dashboard_actions',
-          study_group: 'progress',
-          event: 'view_change_toggle',
-          data_json: JSON.stringify({
-            section_id: this.props.sectionId,
-            old_view: ViewType.SUMMARY,
-            new_view: ViewType.DETAIL
-          })
-        },
-        {includeUserId: true}
-      );
-      this.setState({selectedToggle: ViewType.DETAIL}, () => {
-        setTimeout(() => {
-          this.props.setCurrentView(ViewType.DETAIL);
-        }, 0);
-      });
-    } else {
-      firehoseClient.putRecord(
-        {
-          study: 'teacher_dashboard_actions',
-          study_group: 'progress',
-          event: 'view_change_toggle',
-          data_json: JSON.stringify({
-            section_id: this.props.sectionId,
-            old_view: ViewType.DETAIL,
-            new_view: ViewType.SUMMARY
-          })
-        },
-        {includeUserId: true}
-      );
-      this.setState({selectedToggle: ViewType.SUMMARY}, () => {
+    this.setState(
+      {selectedToggle: selectedToggle, initialView: selectedToggle},
```

Comment:
the existing optimization here is disappointing, since it still leaves the progress panes slow to load. a better optimization might be to render both views and show/hide them via CSS so that they don't take so long to switch (unrendering and rerendering them is much more expensive). Please don't copy this pattern. :-)

Reply:
oh! yikes! Thanks for letting me know.  I'll try to hide/show with CSS.  Will that make the first render of the page take longer though since it will be rendering the content for all three tabs?

Reply:
yes... this could make the initial render take longer. each solution seems to be a trade-off. you could find a way to only render each tab the first time you go to it, but then hide (rather than unrender) each tab when you go away from it, but that would add substantial complexity. 

It looks like https://react-bootstrap.github.io/components/tabs/ has an `unmountOnExit` param you can pass in, which might give us the best of both worlds, so that's worth considering if we go further down this path.